src/dbus_nm.rs

```diff
@@ -0,0 +1,412 @@
+use std::collections::HashMap;
+
+use enum_primitive::FromPrimitive;
+
+use dbus::Path;
+use dbus::arg::{Dict, Variant, Iter, Array, RefArg};
+
+use dbus_api::{DBusApi, extract, utf8_vec_u8_to_string, utf8_variant_to_string,
+               string_to_utf8_vec_u8, path_to_string, VariantTo};
+use manager::{Connectivity, NetworkManagerState};
+use connection::{ConnectionSettings, ConnectionState};
+use device::{DeviceType, DeviceState};
+use wifi::{NM80211ApSecurityFlags, NM80211ApFlags, Security, WEP, NONE};
+
+
+type SettingsMap = HashMap<String, Variant<Box<RefArg>>>;
+
+pub const NM_SERVICE_MANAGER: &'static str = "org.freedesktop.NetworkManager";
+
+pub const NM_SERVICE_PATH: &'static str = "/org/freedesktop/NetworkManager";
+pub const NM_SETTINGS_PATH: &'static str = "/org/freedesktop/NetworkManager/Settings";
+
+pub const NM_SERVICE_INTERFACE: &'static str = "org.freedesktop.NetworkManager";
+pub const NM_SETTINGS_INTERFACE: &'static str = "org.freedesktop.NetworkManager.Settings";
+pub const NM_CONNECTION_INTERFACE: &'static str = "org.freedesktop.NetworkManager.Settings.\
+                                                   Connection";
+pub const NM_ACTIVE_INTERFACE: &'static str = "org.freedesktop.NetworkManager.Connection.Active";
+pub const NM_DEVICE_INTERFACE: &'static str = "org.freedesktop.NetworkManager.Device";
+pub const NM_WIRELESS_INTERFACE: &'static str = "org.freedesktop.NetworkManager.Device.Wireless";
+pub const NM_ACCESS_POINT_INTERFACE: &'static str = "org.freedesktop.NetworkManager.AccessPoint";
+
+pub const NM_WEP_KEY_TYPE_PASSPHRASE: u32 = 2;
+
+
+pub struct DBusNetworkManager {
+    dbus: DBusApi,
+}
+
+impl DBusNetworkManager {
```

Comment:
I gave it a try, but I am giving up on optimizing the paths to use byte strings, because propagating those to the dbus-rs lib we are using makes things really hairy.